crypto/keys/src/ed25519.rs

```diff
@@ -356,6 +386,104 @@ impl Verifier<Ed25519Signature> for Ed25519Pair {
     }
 }
 
+/// An Ed25519 signature.
+#[derive(Copy, Clone, Debug, Deserialize, Serialize)]
+pub struct Ed25519Signature(Signature);
+
+impl Ed25519Signature {
+    /// Create a new signature from a byte array
+    pub fn new(bytes: [u8; SIGNATURE_LENGTH]) -> Self {
+        Self(Signature::from(bytes))
+    }
+
+    /// Return the inner byte array
+    pub fn to_bytes(&self) -> [u8; SIGNATURE_LENGTH] {
+        self.0.to_bytes()
+    }
+}
+
+// Deriving doesn't work, ed25519 crate says:
+// derive `PartialEq` after const generics are available
+impl Eq for Ed25519Signature {}
+
+impl PartialEq for Ed25519Signature {
+    fn eq(&self, other: &Self) -> bool {
+        self.0.eq(&other.0)
+    }
+}
+
+// This is needed to implement prost::Message
+impl Default for Ed25519Signature {
+    fn default() -> Self {
+        Self::new([0; SIGNATURE_LENGTH])
+    }
+}
+
+impl SignatureTrait for Ed25519Signature {
+    fn from_bytes(bytes: &[u8]) -> Result<Self, Error> {
+        Ok(Self(Signature::from_bytes(bytes)?))
+    }
+}
+
+impl AsRef<[u8]> for Ed25519Signature {
+    fn as_ref(&self) -> &[u8] {
+        self.0.as_ref()
+    }
+}
+
+impl<'a> TryFrom<&'a [u8]> for Ed25519Signature {
+    type Error = Ed25519SignatureError;
+
+    fn try_from(bytes: &'a [u8]) -> Result<Self, Error> {
+        Ok(Self(Signature::try_from(bytes)?))
+    }
+}
+
+impl Message for Ed25519Signature {
```

Comment:
The signature is 64 bytes, so we can't use the `prost_message_helper32!` macro...

Reply:
can we use this? https://github.com/mobilecoinofficial/mobilecoin/pull/79

Reply:
there are usage examples in the companion PR: https://github.com/mobilecoinofficial/mobilecoin/pull/74